2_process/src/aggregate_observations.R

```diff
@@ -0,0 +1,150 @@
+  #' Parameter, Value, Value_cd, Value_Max, Value_Max_cd, Value_Min,
+  #' Value_Min_cd, dat_src
+  #' 
+  #' @value returns a data frame containing discrete and continuous 
+  #' water quality observations
+  #' 
+  
+  # Reformat WQP data to match NWIS datasets
+  wqp_data_formatted <- wqp_data %>%
+    # format site names for NWIS sites with prefix "USGS"
+    mutate(site_id = if_else(grepl('USGS',MonitoringLocationIdentifier),
+                             substr(MonitoringLocationIdentifier, 6, 100),
+                             MonitoringLocationIdentifier),
+           Parameter = "SpecCond") %>%
+    # harmonize column names with cols used in nwis_data
+    rename(agency_cd = OrganizationIdentifier,
+           site_no = site_id,
+           Date = ActivityStartDate) %>%
+    # find the min/max and the mean (if multiple observations exist 
+    # for a unique site-date)
+    group_by(site_no, Date, agency_cd, Parameter) %>%
+    summarize(Value = round(mean(resultVal2, na.rm = TRUE), 1),
+              Value_cd = unique(ResultStatusIdentifier),
+              Value_Max = max(resultVal2, na.rm = TRUE),
+              Value_Max_cd = NA_character_,
+              Value_Min = min(resultVal2, na.rm = TRUE),
+              Value_Min_cd = NA_character_,
+              .groups = "keep") %>%
+    ungroup() %>%
+    # pare down columns of interest and specify the data source
+    select(agency_cd, site_no, Date, Parameter, Value, Value_cd, 
+           Value_Max, Value_Max_cd, Value_Min, Value_Min_cd) %>%
+    mutate(dat_src = "WQP")
+  
+  # Combine wqp_data and nwis_data into a single data frame
+  nwis_data$dat_src <- "NWIS"
+  
+  combined_data <- bind_rows(wqp_data_formatted, nwis_data)
+  
+  return(combined_data) 
+  
+}
+
+
+
+aggregate_observations <- function(wqp_data, nwis_data, sites_w_segs, 
+                                   aggr_method = "reach", prefer_nwis_sites = FALSE){
+  #' 
+  #' @description Function to aggregate water quality observations to 1 value/day
+  #'
+  #' @param wqp_data data frame containing discrete water quality data from the wQP. 
+  #' Must contain cols MonitoringLocationIdentifier, OrganizationIdentifier, 
+  #' ActivityStartDate, resultVal2, ResultStatusIdentifier
+  #' @param nwis_data data frame containing combined instantaneous (uv) and daily (dv) data; 
+  #' expected cols are agency_cd, site_no, Date, Parameter, Value, Value_cd, Value_Max, 
+  #' Value_Max_cd, Value_Min, Value_Min_cd, dat_src
+  #' @param sites_w_segs data frame containing site locations and matched segment id's;
+  #' must contain column site_id
+  #' @param aggr_method character vector indicating how observations should be aggregated;
+  #' options include "reach" and "site" (defaults to "reach"). If "reach", then values 
+  #' returned represent 1 value/segment/day, if "site", then values will be returned for
+  #' each lat/lon location (maintaining multiple sites along a segment) amounting to 
+  #' 1 value/site/day
```

Comment:
Could you say why we need this? Something like this (is this true)??
```suggestion
  #' 1 value/site/day. Aggregating by "site" can be useful because occasionally there are multiple sites that have the same lat and lon. 
```

Reply:
Yeah, good idea! I added some text in the documentation here to explain that a user might just want to retain the unaggregated, raw data but also, we aggregate by 'site' because there are sometimes multiple site names for the same lat/lon.